File: Codex/Python/2025/Day1_Solution_2025.py
def main() -> None:
    position = 50
    zero_count = 0

    with open("Day1_Input.txt", "r", encoding="utf-8") as file:
        for raw_line in file:
            line = raw_line.strip()
            if not line:
                continue

            direction = line[0]
            distance = int(line[1:]) % 100

            if direction == "L":
                position = (position - distance) % 100
            else:
                position = (position + distance) % 100

            if position == 0:
                zero_count += 1

    print(zero_count)


if __name__ == "__main__":
    main()

File: Codex/Rust/2025/Day1_Solution_2025.rs
use std::fs;

fn main() {
    let input = fs::read_to_string("Day1_Input.txt").expect("failed to read Day1_Input.txt");

    let mut position: i64 = 50;
    let mut count = 0_u64;

    for line in input.lines().filter(|line| !line.trim().is_empty()) {
        let direction = line.as_bytes()[0];
        let distance: i64 = line[1..]
            .trim()
            .parse()
            .expect("invalid rotation distance");

        let step = distance.rem_euclid(100);
        position = match direction {
            b'L' => (position - step).rem_euclid(100),
            b'R' => (position + step).rem_euclid(100),
            _ => panic!("invalid rotation direction"),
        };

        if position == 0 {
            count += 1;
        }
    }

    println!("{count}");
}
